homeassistant/components/matter/lock.py

```diff
@@ -120,9 +134,14 @@ async def async_unlock(self, **kwargs: Any) -> None:
 
     async def async_open(self, **kwargs: Any) -> None:
         """Open the door latch."""
-        # optimistically signal unlocking to state machine
-        self._attr_is_unlocking = True
+        # optimistically signal opening to state machine
+        self._attr_is_opening = True
         self.async_write_ha_state()
+        # the lock should acknowledge the command with an attribute update
+        # but bad things may happen, so guard against it with a timer.
+        self._optimistic_timer = self.hass.loop.call_later(
```

Comment:
We have the event helper `async_call_later`.

Reply:
what's the benefit of that ?

Reply:
It's an abstraction over the low level event loop API. Generally we want to use our helpers when we can.

Reply:
I have a follow-up PR open, I can use the helper right away

Reply:
![image](https://github.com/home-assistant/core/assets/6389780/616842fc-efb7-4d29-98e4-0954038e401a)

![image](https://github.com/home-assistant/core/assets/6389780/5d46e055-41fb-452a-8178-4c6a83a4557f)

![image](https://github.com/home-assistant/core/assets/6389780/0bec3dbf-b2fd-45cc-8cd4-f1d38b451363)

Reply:
The callback should accept a single datetime parameter.

Reply:
I just want to mimic the behavior of loop.call_later --> call a callback method after X period of time.
It does work with that helper but mypy chokes